src/ow1/heroes/baptiste.opy

```diff
@@ -29,17 +38,79 @@ rule "[baptiste.opy]: Set default immortality field cooldown":
     waitUntil(not eventPlayer.isUsingAbility2(), 9999)
     eventPlayer.setAbilityCooldown(Button.ABILITY_2, OW1_BAPTISTE_IMMORTALITY_COOLDOWN)
 
+rule "[baptiste.opy]: Show exo boots charge bar":
+    @Event eachPlayer
+    @Hero baptiste
+    @Condition eventPlayer.exo_boots_charge_normalized > 0
+    
+    
+
+rule "[baptiste.opy]: Hide exo boots charge bar":
+    @Event eachPlayer
+    @Hero baptiste
+    @Condition eventPlayer.exo_boots_charge_normalized == 0
+
+    
+
+rule "[baptiste.opy]: Start charging exo boots":
+    @Event eachPlayer
+    @Hero baptiste
+    @Condition eventPlayer.isCrouching()
+
+    eventPlayer.exo_boots_hud_visible = true
+    # don't allow the player to jump normally, because that would use the OW2 exo boots which charge faster
+    eventPlayer.setJumpEnabled(false)
+    chase(eventPlayer.exo_boots_charge_normalized, 1, duration=OW1_BAPTISTE_EXO_BOOTS_CHARGE_TIME, ChaseReeval.NONE)
+
+rule "[baptiste.opy]: Handle exo boots charge reset":
+    @Event eachPlayer
+    @Hero baptiste
+    @Condition not eventPlayer.isCrouching()
+    @Condition eventPlayer.exo_boots_charge_normalized > 0
+
+    # stop charging exo boots
+    stopChasingVariable(eventPlayer.exo_boots_charge_normalized)
+    # if the player crouches again, or jumps, or the linger time runs out, the exo boots charge goes back to 0
+    waitUntil(eventPlayer.isCrouching() or (eventPlayer.isOnGround() and eventPlayer.isHoldingButton(Button.JUMP)), OW1_BAPTISTE_EXO_BOOTS_LINGER_DURATION)
+    eventPlayer.exo_boots_hud_visible = false
+    # another grace period in case the player chose to jump, so we can use exo_boots_charge_normalized for that
+    wait()
+    eventPlayer.exo_boots_charge_normalized = 0
+    eventPlayer.setJumpEnabled(true)
+
+
+# TODO: the speed here needs to be tuned because this doesn't account for normal jump height without exo boots
```

Comment:
Do you know what the speed for Baptiste's default jump is or where I can find that? I think I just need to add to the base jump speed (and scale accordingly) but don't know exactly what it is

Reply:
for a bit more clarification, when he jumps with very little charge he barely even leaves the ground but he should be going at least as high as a normal jump.